### PR TITLE
fix(testing): define constructor

### LINKED ISSUE #583: createMock is not creating/providing `constructor`
We are trying to use `createMock` to mock some of our dependencies in integration test.

We are using it like:
```ts
  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    })
      .overrideProvider(AwsS3Service)
      .useValue(
        createMock<AwsS3Service>({
          downloadAsString: jest.fn().mockResolvedValue('{}'),
        }),
      )
      .compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });
```

But we get the following error:
```ts
 FAIL  test/integration/app.spec.ts (6.267 s)
  AppController (integration)
    ✕ / (GET) (1362 ms)

  ● AppController (integration) › / (GET)

    TypeError:

      40 |
      41 |     app = moduleFixture.createNestApplication();
    > 42 |     await app.init();
         |     ^
      43 |   });
      44 |
      45 |   afterEach(async () => {

      at Reflect.getMetadata (node_modules/reflect-metadata/Reflect.js:354:23)
      at PluginsExplorerService.filterPlugins (node_modules/@nestjs/apollo/dist/services/plugins-explorer.service.js:20:34)
      at node_modules/@nestjs/apollo/dist/services/plugins-explorer.service.js:13:57
      at node_modules/@nestjs/graphql/dist/services/base-explorer.service.js:30:51
          at Array.map (<anonymous>)
      at node_modules/@nestjs/graphql/dist/services/base-explorer.service.js:30:34
          at Array.map (<anonymous>)
      at invokeMap (node_modules/@nestjs/graphql/dist/services/base-explorer.service.js:28:28)
      at PluginsExplorerService.flatMap (node_modules/@nestjs/graphql/dist/services/base-explorer.service.js:33:42)
      at PluginsExplorerService.explore (node_modules/@nestjs/apollo/dist/services/plugins-explorer.service.js:13:21)
      at ApolloDriver.start (node_modules/@nestjs/apollo/dist/drivers/apollo.driver.js:18:112)
      at GraphQLModule.onModuleInit (node_modules/@nestjs/graphql/dist/graphql.module.js:105:36)
      at callModuleInitHook (node_modules/@nestjs/core/hooks/on-module-init.hook.js:51:9)
      at Proxy.callInitHook (node_modules/@nestjs/core/nest-application-context.js:224:13)
      at Proxy.init (node_modules/@nestjs/core/nest-application.js:98:9)
      at Object.<anonymous> (test/integration/app.spec.ts:42:5)
```

Which points me towards this line https://github.com/nestjs/graphql/blob/v10.2.1/packages/apollo/lib/services/plugins-explorer.service.ts#L24 which is trying to read `instance.constructor` which is not present on `createMock()` objects.

This causes the `reflect-metadata` call to fail with `TypeError:` https://github.com/rbuckton/reflect-metadata/blob/master/Reflect.js#L352-L358

Is it possible to provide the `instance.constructor` somehow?

### PR DESCRIPTION
**Pull request by mentos1386**
Fixes https://github.com/golevelup/nestjs/issues/583

### CODE REVIEW

File: packages/testing/src/mocks.ts [hunk 1]
```diff
@@ -66,8 +67,11 @@ export const createMock = <T extends object>(
 
   const proxy = new Proxy(partial, {
     get: (obj, prop) => {
+      if (prop === 'constructor') {
+        return () => undefined;
+      }
```
Comment (mentos1386) [inline packages/testing/src/mocks.ts:72]: This solves our issue. But i'm not sure if it doesn't break anything else.

Is there a better way to solve this?

**Comment by mentos1386** [comment]
Hey @WonderPanda, do you maybe have some time to look in to this?

File: packages/testing/src/mocks.ts (hunk 1 above)
Comment (WonderPanda) [reply packages/testing/src/mocks.ts:72]: This seems okay for right now. It didn't break any existing tests so we can merge and see if there are any reports from the community

**Comment by WonderPanda** [comment]
Published in @golevelup/ts-jest@0.3.6

File: packages/testing/src/mocks.ts (hunk 1 above)
Comment (regevbr) [reply packages/testing/src/mocks.ts:72]: see #587

**Pull request merged**